core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -415,6 +415,8 @@ public Path getWorkingDirectory() {
   @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
   @Override
   public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throws IOException {
+    Preconditions.checkArgument(uri.getScheme().equals(getScheme()),
```

Comment:
can `uri.getScheme()` be `null`? we don't want a `NullPtrException` thrown here at least

Reply:
it shouldn't be `null`, HDFS chooses which implementation of `FileSystem` to use based on the scheme.